### PR TITLE
[DataFactory]Added new features into 4.16.0

### PR DESCRIPTION
**Pull request by Jingshu923**
- Added Amazon S3 Compatible and Oracle Cloud Storage connectors

Azure/azure-rest-api-specs#13368

- Updated AzureDataExplorer LinkedService servicePrincipalId, servicePrincipalKey, tenant properties from required to optional

- Added accountKind property to AzureBlobStorage LinkedService

Azure/azure-rest-api-specs#13291

- Added AAD auth to CosmosDb LinkedService

Azure/azure-rest-api-specs#13257

- Added authHeaders to HTTP/REST/OData LinkedService

- Added MultiFactor auth to SFTP LinkedService
Azure/azure-rest-api-specs#13204

- Added UAMI to Identity: Azure/azure-rest-api-specs#13174

- Added privateEndpointConnection apis to DataFactory: Azure/azure-rest-api-specs#13249

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Generated/Models/FactoryIdentity.cs [hunk 1]
```diff
@@ -31,31 +32,35 @@ public FactoryIdentity()
...
+        /// <param name="type">The identity type. Possible values include:
+        /// 'SystemAssigned', 'UserAssigned',
+        /// 'SystemAssigned,UserAssigned'</param>
         /// <param name="principalId">The principal id of the identity.</param>
         /// <param name="tenantId">The client tenant id of the
         /// identity.</param>
         /// <param name="userAssignedIdentities">List of user assigned
         /// identities for the factory.</param>
-        public FactoryIdentity(System.Guid? principalId = default(System.Guid?), System.Guid? tenantId = default(System.Guid?), IDictionary<string, object> userAssignedIdentities = default(IDictionary<string, object>))
+        public FactoryIdentity(string type, System.Guid? principalId = default(System.Guid?), System.Guid? tenantId = default(System.Guid?), IDictionary<string, object> userAssignedIdentities = default(IDictionary<string, object>))
```
Comment (markcowl) [inline sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Generated/Models/FactoryIdentity.cs:43]: Breaking change - you should include a customization that implements the old method signature as an overload for this.  This is simply adding a partial class in the customizations folder with the old constructor calling the new constructor.

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Generated/Models/AzureDataExplorerLinkedService.cs [hunk 2]
```diff
@@ -41,24 +41,24 @@ public AzureDataExplorerLinkedService()
...
+        /// <param name="servicePrincipalId">The ID of the service principal
+        /// used to authenticate against Azure Data Explorer. Type: string (or
+        /// Expression with resultType string).</param>
+        /// <param name="servicePrincipalKey">The key of the service principal
+        /// used to authenticate against Kusto.</param>
+        /// <param name="tenant">The name or ID of the tenant to which the
+        /// service principal belongs. Type: string (or Expression with
+        /// resultType string).</param>
+        public AzureDataExplorerLinkedService(object endpoint, object database, IDictionary<string, object> additionalProperties = default(IDictionary<string, object>), IntegrationRuntimeReference connectVia = default(IntegrationRuntimeReference), string description = default(string), IDictionary<string, ParameterSpecification> parameters = default(IDictionary<string, ParameterSpecification>), IList<object> annotations = default(IList<object>), object servicePrincipalId = default(object), SecretBase servicePrincipalKey = default(SecretBase), object tenant = default(object))
```
Comment (markcowl) [inline sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Generated/Models/AzureDataExplorerLinkedService.cs:61]: This is a breaking change.  You should include a customization that implements the old contructor signature.

**Review by markcowl — CHANGES_REQUESTED** [review]
A couple of customizations to prevent breaking changes, also, please update AzSDK.RP.Props with the new api-version

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Generated/Models/AzureDataExplorerLinkedService.cs (hunk 2 above)
Comment (Jingshu923) [reply sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Generated/Models/AzureDataExplorerLinkedService.cs:61]: Add class /src/Extensions/AzureDataExplorerLinkedService.cs

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Generated/Models/FactoryIdentity.cs (hunk 1 above)
Comment (Jingshu923) [reply sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Generated/Models/FactoryIdentity.cs:43]: Add class /src/Extensions/FactoryIdentity.cs

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Extensions/AzureDataExplorerLinkedService.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
+﻿using System;
```
Comment (markcowl) [inline sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Extensions/AzureDataExplorerLinkedService.cs:1]: By convention, these should go in the 'Customizations' folder

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Extensions/FactoryIdentity.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
```
Comment (markcowl) [inline sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Extensions/FactoryIdentity.cs:1]: By convention, this should go in the 'Customizations' folder

**Review by markcowl — CHANGES_REQUESTED** [review]

**Review by markcowl — APPROVED** [review]

**Review by markcowl — APPROVED** [review]

**Pull request merged**